lib/matplotlib/cycle.py

```diff
@@ -0,0 +1,140 @@
+import itertools
+import six
+from matplotlib import rcParams
+
+
+class Cycle:
+
+    def __init__(self):
+        """
+        Set the initial cycle of styles to be used by the lines of the graph
+        """
+        self._styles = {'color': None,
+                        'linestyle': None,
+                        'linewidth': None,
+                        'marker': None,
+                        'markersize': None,
+                        'markeredgewidth': None,
+                        'markeredgecolor': None,
+                        'markerfacecolor': None,
+                        'antialiased': None,
+                        'dash_capstyle': None,
+                        'solid_capstyle': None,
+                        'dash_joinstyle': None,
+                        'solid_joinstyle': None,
+                        }
+        self._styles_list = {}
+        self.set_color_cycle()
+        self.set_line_cycle()
+
+    def __getstate__(self):
+        return {'_styles_list': self._styles_list}
+
+    def __setstate__(self, state):
+        self.__init__()
+        self.__dict__.update(state)
+        for style in self._styles_list.keys():
+            self.set_cycle(style, self._styles_list[style])
+
+    def next(self, args={}):
```

Comment:
Don't use mutable objects as default values (they only get instantiated once at import time and can lead to some _really_ strange bugs or some semi-evil, but kinda fun, features).